lntest/itest/lnd_funding_test.go

```diff
@@ -744,3 +747,120 @@ func deriveFundingShim(net *lntest.NetworkHarness, t *harnessTest,
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+	net.SendCoins(t.t, btcutil.SatoshiPerBitcoin, alice)
+
+	bob := net.NewNode(t.t, "bob", commitTypeTweakless.Args())
+	defer shutdownAndAssert(net, t, bob)
+
+	chanAmt := btcutil.Amount(150000)
+	pushAmt := btcutil.Amount(0)
+
+	net.ConnectNodes(t.t, alice, bob)
+
+	// Check existing connection.
+	assertNumConnections(t, alice, bob, 1)
+
+	pendingUpdate, err := net.OpenPendingChannel(alice, bob, chanAmt, pushAmt)
+
+	assertNumOpenChannelsPending(t, alice, bob, 1)
+	require.NoError(t.t, err, "error opening pending channel")
+
+	fundingTxHash, _ := chainhash.NewHash(pendingUpdate.Txid)
+	fundingTx, err := net.Miner.Client.GetRawTransaction(fundingTxHash)
+	require.NoError(t.t, err, "error fetching raw tx")
+
+	// Fetch the tx which funds the fundingTx so we can collect it's outpoint
+	previousOutPoint := &fundingTx.MsgTx().TxIn[0].PreviousOutPoint.Hash
+	previousOutPointTx, err := net.Miner.Client.GetRawTransaction(previousOutPoint)
+	require.NoError(t.t, err, "error fetching previousOutPoint tx")
+
+	// Create a double spend tx which will spend from the outpoints that funded the
+	// funding tx
+	txDoubleSpend := wire.NewMsgTx(2)
+
+	// Add the fundingTx input's to the double spend transaction
+	txDoubleSpend.AddTxIn(fundingTx.MsgTx().TxIn[0])
+
+	addrReq := &lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_WITNESS_PUBKEY_HASH,
+	}
+
+	newAddressResp, err := alice.NewAddress(ctxb, addrReq)
+	require.NoError(t.t, err, "error new address")
+
+	addr, err := btcutil.DecodeAddress(newAddressResp.Address, net.Miner.ActiveNet)
+	require.NoError(t.t, err, "error decode address")
+
+	addrScript, err := txscript.PayToAddrScript(addr)
+	require.NoError(t.t, err, "error pay address")
+
+	output := &wire.TxOut{
+		PkScript: addrScript,
+		Value:    int64(2000),
+	}
+
+	txDoubleSpend.AddTxOut(output)
+	txDoubleSpendBytesBuffer := bytes.Buffer{}
+	err = txDoubleSpend.Serialize(&txDoubleSpendBytesBuffer)
+	require.NoError(t.t, err, "error serializing tx")
+
+	// Add the outpoints to the SignDescriptor to sign the double spend tx
+	signDescription := signrpc.SignDescriptor{
+		Output: &signrpc.TxOut{
+			Value:    previousOutPointTx.MsgTx().TxOut[0].Value,
+			PkScript: previousOutPointTx.MsgTx().TxOut[0].PkScript,
+		},
+		WitnessScript: nil,
+		InputIndex:    0,
+	}
+
+	signResponse, err := alice.SignerClient.ComputeInputScript(ctxb, &signrpc.SignReq{
+		SignDescs:  []*signrpc.SignDescriptor{&signDescription},
+		RawTxBytes: txDoubleSpendBytesBuffer.Bytes(),
+	})
+	require.NoError(t.t, err, "error signing tx")
+
+	// Add the witness and spend the transaction
+	txDoubleSpend.TxIn[0].Witness = signResponse.InputScripts[0].Witness
+	txs := []*btcutil.Tx{btcutil.NewTx(txDoubleSpend)}
+	_, err = net.Miner.GenerateAndSubmitBlock(txs, -1, time.Time{})
+	require.NoError(t.t, err, "error submitting tx")
+	// Generate some more blocks so Alice can detect the double spend
+	_, err = net.Miner.Client.Generate(10)
```

Comment:
This tests that the transaction is considered double-spent after 11 confirmations. Is there a less arbitrary looking number? It could be user-configurable or just use the same value as for confirming a channel. Then the test should test that the configured value matches. (Opening is not canceled before enough confirmations and is after.)

Reply:
These integration tests aren't really user-configurable

Reply:
The value in the test is not related to any fixed variable in the implementation. The double spend handling logic kicks in whenever a spend is detected on the chain after the funding broadcast height.

https://github.com/johng/lnd/blob/030abf6fdfb6b7dc40dcccf843b5d2be20a03194/funding/manager.go#L1084-L1087

The test set this arbitrary value to ensure it was detected.

Reply:
@Crypt-iQ I'm talking about the value itself, not the test.

> whenever a spend is detected on the chain after the funding broadcast height

This looks like the correct, non-arbitrary value.
Shouldn't the test check that this specific value is obeyed? IOW test that it's **not** considered double-spent one block prior to it and is one block after.